chia/rpc/wallet_rpc_api.py

```diff
@@ -1685,6 +1687,74 @@ async def did_message_spend(self, request) -> EndpointResult:
         spend_bundle = await wallet.create_message_spend(coin_announcements, puzzle_announcements)
         return {"success": True, "spend_bundle": spend_bundle}
 
+    async def did_get_info(self, request) -> EndpointResult:
+        if "coin_id" not in request:
+            return {"success": False, "error": "Coin ID is required."}
+        coin_id = request["coin_id"]
+        if coin_id.startswith(AddressType.DID.hrp(self.service.config)):
+            coin_id = decode_puzzle_hash(coin_id)
+        else:
+            coin_id = bytes32.from_hexstr(coin_id)
+        # Get coin state
+        peer: Optional[WSChiaConnection] = self.service.get_full_node_peer()
+        if peer is None:
+            raise ValueError("No peers to get info from")
+        coin_state_list: List[CoinState] = await self.service.wallet_state_manager.wallet_node.get_coin_state(
+            [coin_id], peer=peer
+        )
+        if coin_state_list is None or len(coin_state_list) < 1:
+            return {"success": False, "error": f"Coin record 0x{coin_id.hex()} not found"}
+        coin_state: CoinState = coin_state_list[0]
```

Comment:
This chunk of code looks identical to the same functionality found in `nft_get_info`. Can this be refactored into a common method that both RPCs can use?

Maybe there's more of this RPC and `nft_get_info` that can be refactored?